src/main/java/com/liferay/faces/util/cache/CacheFactory.java

```diff
@@ -0,0 +1,228 @@
+ */
+public abstract class CacheFactory implements Wrapper<CacheFactory> {
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be used during execution of a request thread, so it
+	 * is not guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context. It is needed in order
+	 *                          for the {@link FactoryExtensionFinder} to be able to find the factory.
+	 */
+	public static <K, V> Cache<K, V> getCacheInstance(ExternalContext externalContext) {
+
+		CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+				CacheFactory.class);
+
+		return cacheFactory.<K, V>getCache();
+	}
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be used during execution of a request thread, so it
+	 * is not guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context. It is needed in order
+	 *                          for the {@link FactoryExtensionFinder} to be able to find the factory.
+	 * @param  initialCapacity  The initial capacity of the cache.
+	 */
+	public static <K, V> Cache<K, V> getCacheInstance(ExternalContext externalContext, int initialCapacity) {
+
+		CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+				CacheFactory.class);
+
+		return cacheFactory.<K, V>getCache(initialCapacity);
+	}
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be used during execution of a request thread, so it
+	 * is not guaranteed to be {@link java.io.Serializable}. The returned cache will avoid exceeding the maximum cache
+	 * capacity. The algorithm used to ensure that the cache size does not exceed the max capacity (for example
+	 * least-recently-used) may cause significant performance differences between the returned cache and a cache
+	 * returned from {@link #getCache()} or {@link #getCache(int)}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context. It is needed in order
+	 *                          for the {@link FactoryExtensionFinder} to be able to find the factory.
+	 * @param  initialCapacity  The initial capacity of the cache.
```

Comment:
@stiemannkj1 Just realized that for `initialCapacity` we spell out the word "initial" but for `maxCapacity` we use "max" as an abbreviation rather than "maximum" -- Not requesting a change, but I noticed an inconsistency and just thought I'd bring it up.